### ISSUE TITLE
I was looking for a pattern that contains ".mint" files, such as "source/**/*.mint", Got "...." when running mint format.

### ISSUE THREAD

**Issue by zw963** [mint-lang/mint#549]
I am try to fixing format buffer feature use `mint format` in the [emacs-mint-mode](https://github.com/creatorrr/emacs-mint-mode) issue.

After some debug, i get following error message instead.

```
Mint - Formatting files[
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
I was looking for a pattern that contains ".mint" files,
such as "source/**/*.mint". Got "/home/common/Study/Mint/test_formatter/source/Main.mint" instead.
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
All done in 3.502ms!
```

It wired, because, because sure i can run command successful in my konsole.

```sh
 ╭─ 21:07  zw963 ⮀ ~/Study/Mint/test_formatter ⮀ ⭠ (30f376c) master *%  ➦ ruby-3.1.0 
 ╰─ $ mint format  /home/common/Study/Mint/test_formatter/source/Main.mint
Mint - Formatting files
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
Formatted: /home/common/Study/Mint/test_formatter/source/Main.mint
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
All done in 314μs!
 ╰─ $ 1  pwd
/home/zw963/Study/Mint/test_formatter
```

The above issue will happen if current PWD  directory is not the project root

```
 ╰─ $ cd ~

 ╭─ 21:10  zw963 ⮀ ~ ⮀ ➦ ruby-3.1.0  0 
 ╰─ $ mint format  /home/common/Study/Mint/test_formatter/source/Main.mint
Mint - Formatting files
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
I was looking for a pattern that contains ".mint" files,
such as "source/**/*.mint". Got "/home/common/Study/Mint/test_formatter/source/Main.mint" instead.
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
All done in 3.554ms!
```

In this case, mint format never format file.

So, the really issue is,  we need a way to told formatter, where is the project root.

Following is some solution:

1. like crystal tool format, i can run any file outside project root.
2. mint format provided a command line option for specify the path of mint.json, then, when formatter do format,
    can pass this argument, for my case,  format can do like this:
    mint format --config /home/common/Study/Mint/test_formatter/ /home/common/Study/Mint/test_formatter/source/Main.mint
3. Or support set a environment variable, where project root is, `mint format` can read it.

**Comment by zw963** [comment]
I workaround this issue in [this](https://github.com/creatorrr/emacs-mint-mode/pull/6/files#diff-9fff452ef6e9978667b351b4b78de702da8682565b1ffdf2947c4298d2baf575R233) PR with:

1. locate the project root with mint.json
2.  goto project root, set this directory as default directory.
3. run code here.

**Comment by zw963** [comment]
Hi,  i get a similiar issue when try to add mint-lang support to [apheleia](https://github.com/radian-software/apheleia)

As you can see, i added support for Crystal is in recent days, it quite easy, check [this pr](https://github.com/radian-software/apheleia/pull/137), and, it support almost all languages for now, even, nix.

But, when i try to add mint support for it, i get same(type) error as this.

```
Sat Oct 29 02:16:33 2022 :: /home/zw963/Crystal/play/mint-realworld-rework/source/
$ mint format --stdin

Unhandled exception: Error opening file with mode 'r': '/home/zw963/Crystal/play/mint-realworld-rework/source/mint.json': No such file or directory (File::NotFoundError)
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???

Command failed with exit code 1.
```

Yes, the checker could not found the mint.json in the same folder of the opened file buffer, most other language format checker can handle this, but, always limit by this when i try to add auto save formatter for mint.

Thanks.

**Comment by gdotdesign** [comment]
I believe this is fixed in `0.20.0`. Please reopen this issue if it's still relevant.
